### PR TITLE
RDM-12309 Make getJurisdictions() access controlled

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-12309

### Change description ###
Updated the JurisdictionsResolver to support access control.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        this.roleAssignmentService = roleAssignmentService;
+    }
+
+    @Override
+    public List<String> getJurisdictions() {
+        String userId = this.userRepository.getUser().getId();
+        RoleAssignments roleAssignments = roleAssignmentService.getRoleAssignments(userId);
+
+        return roleAssignments.getRoleAssignments().stream()
+            .map(RoleAssignment::getAttributes)
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java:39]: is it better to move this to RoleAssigments this logic

File: src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Qualifier(AccessControlledJurisdictionsResolver.QUALIFIER)
+@RequestScope
+public class AccessControlledJurisdictionsResolver implements JurisdictionsResolver, AccessControl {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java:20]: Can we save AttributeBasedJurisdictionResolver?

**Review by kiran-yenigala-hmcts — APPROVED** [review]
minor Comments

File: src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java (hunk 1 above)
Comment (rafalkalita) [reply src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java]: done

File: src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java (hunk 2 above)
Comment (rafalkalita) [reply src/main/java/uk/gov/hmcts/ccd/data/user/AccessControlledJurisdictionsResolver.java]: done

**Pull request merged**